File: SOLID-Java/src/main/java/openclose/readwriter/DOCReaderWriter.java
package openclose.readwriter;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.net.URL;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.logging.Logger;
import java.util.stream.Collectors;

public class DOCReaderWriter implements FileReaderWriter
{
	private final static Logger LOGGER = Logger.getLogger(DOCReaderWriter.class.getName());
	
	public List<String> read()
	{
		List<String> fileNames = new ArrayList<>();
		LOGGER.info("Reading DOCs");
		final URL resource = getClass().getResource("/docs");
		final File file = new File(resource.getPath());
		if (file.exists() && file.isDirectory())
		{
			final List<File> files = Arrays.asList(file.listFiles());
			files.sort((File o1, File o2) -> o1.getName().compareTo(o2.getName()));
			fileNames = files.stream().map(x -> x.getName()).collect(Collectors.toList());
		}
		return fileNames;
	}

	public void write()
	{
		try
		{
			final File file = new File("resource/docs/MyDOC.doc");
			if (file.createNewFile())
			{
				LOGGER.info("File is created!");
			} else
			{
				LOGGER.info("File already exists.");
			}

			final FileWriter writer = new FileWriter(file);
			writer.write("Test doc");
			writer.close();
		} catch (IOException e)
		{
			e.printStackTrace();
		}
	}
}

File: SOLID-Scala/src/main/scala/openclose/readwriter/DOCReaderWriter.scala
package openclose.readwriter

import java.io.File
import java.io.IOException
import java.util.logging.Level
import java.io.FileWriter
import java.util.logging.Level
import java.util.logging.Logger
import scala.util.control.NonFatal
import scala.util.Try

class DOCReaderWriter extends FileReaderWriter{
  private val LOGGER = Logger.getLogger(this.getClass.getName)

  @Override
  def read(): List[String] = {
    val resource = getClass.getResource("/docs")
    val file = new File(resource.getPath)
    if (file.exists() && file.isDirectory) {
      LOGGER.info("Reading DOCs")
      val files = file.listFiles().sortBy { x => x.getName }
      files.map(x => x.getName).toList
    } else {
      List.empty
    }
  }

  @Override
  def write(fileName: String): Unit = {
    var writer: FileWriter = null
    val path = "src/main/resources/"
    try {
      val file = new File(path + s"docs/${fileName}.doc")
      if (file.createNewFile()) {
        LOGGER.info("File is created!")
      } else {
        LOGGER.info("File already exists.")
      }
      writer = new FileWriter(file)
      writer.write("Test doc")
    } catch {
      case NonFatal(e) => LOGGER.log(Level.SEVERE, e.getMessage)
    } finally {
      if (writer != null) {
        writer.close
      }
    }
  }
}